src/serde.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024, Horizen Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloc::{string::String, vec::Vec};
+use proof_of_sql::{
+    base::{
+        database::{OwnedColumn, OwnedTable},
+        math::decimal::Precision,
+    },
+    proof_primitive::dory::DoryScalar,
+    sql::proof::QueryData,
+};
+use proof_of_sql_parser::{
+    posql_time::{PoSQLTimeUnit, PoSQLTimeZone},
+    Identifier,
+};
+use serde::{Deserialize, Serialize};
+
+pub type IndexMap = indexmap::IndexMap<
+    Identifier,
+    OwnedColumn<DoryScalar>,
+    core::hash::BuildHasherDefault<ahash::AHasher>,
+>;
+
+#[derive(Serialize, Deserialize)]
+#[serde(remote = "QueryData<DoryScalar>")]
+pub(crate) struct QueryDataDef {
+    #[serde(with = "OwnedTableDef")]
+    table: OwnedTable<DoryScalar>,
+    verification_hash: [u8; 32],
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(remote = "OwnedTable<DoryScalar>")]
+pub struct OwnedTableDef {
+    #[serde(getter = "OwnedTable::inner_table", with = "index_map_serde")]
+    table: IndexMap,
+}
```

Comment:
Sure? You really need it? Seams that `indexmap` crate already have a `serde` feature. Did you have tried it?